### ISSUE TITLE
V3 LeagueEntry errors when calling attributes

### ISSUE THREAD

**Issue by SuperFrosty** [meraki-analytics/cassiopeia#123]
Made a test.py to call various attributes of league entries, some error, some do not, here's a list of the attributes and their errors.

Firstly, the test code is just:
```python
from cassiopeia.core import Summoner

summoner = Summoner(name="Riviere", region="OCE")
leagues = summoner.leagues
for entry in leagues.fives:
    print(entry.division)
    print(entry.fresh_blood)
    print(entry.hot_streak)
    print(entry.league_points)
    print(entry.losses)
    print(entry.name)
    print(entry.platform)
    print(entry.promos)
    print(entry.queue)
    print(entry.region)
    print(entry.tier)
    print(entry.veteran)
    print(entry.wins)
```
entry.division - works
entry.fresh_blood - errors
```bash
Traceback (most recent call last):
  File "C:\Users\redacted\OneDrive\Bots\BlitzcrankBot\test.py", line 7, in <module>
    print(entry.fresh_blood)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 238, in fresh_blood
    return self._data[LeagueItemData].fresh_blood
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 98, in fresh_blood
    return self._dto["fresh_blood"]
KeyError: 'fresh_blood'
```
entry.hot_streak - works
entry.league_points - works
entry.losses - works
entry.name - errors
```bash
Traceback (most recent call last):
  File "C:\Users\redacted\OneDrive\Bots\BlitzcrankBot\test.py", line 11, in <module>
    print(entry.name)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 206, in name
    return self._data[LeagueItemData].name
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 58, in name
    return self._dto["leagueName"]
KeyError: 'leagueName'
```
entry.platform - errors
```bash
Traceback (most recent call last):
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 16, in wrapper
    return values[self]
  File "C:\Users\redacted\Anaconda3\lib\weakref.py", line 394, in __getitem__
    return self.data[ref(key)]
KeyError: <weakref at 0x00000141C9936458; to 'LeagueEntry' at 0x00000141C995AFD0>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 16, in wrapper
    return values[self]
  File "C:\Users\redacted\Anaconda3\lib\weakref.py", line 394, in __getitem__
    return self.data[ref(key)]
KeyError: <weakref at 0x00000141C9936458; to 'LeagueEntry' at 0x00000141C995AFD0>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\redacted\OneDrive\Bots\BlitzcrankBot\test.py", line 12, in <module>
    print(entry.platform)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 18, in wrapper
    values[self] = method(self)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 194, in platform
    return self.region.platform
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 18, in wrapper
    values[self] = method(self)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 189, in region
    return Region(self._data[LeagueItemData].region)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 46, in region
    return self._dto["region"]
KeyError: 'region'
```
entry.promos - errors
```bash
Traceback (most recent call last):
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 16, in wrapper
    return values[self]
  File "C:\Users\redacted\Anaconda3\lib\weakref.py", line 394, in __getitem__
    return self.data[ref(key)]
KeyError: <weakref at 0x000001F656426458; to 'LeagueEntry' at 0x000001F65644AFD0>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\redacted\OneDrive\Bots\BlitzcrankBot\test.py", line 13, in <module>
    print(entry.promos)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 18, in wrapper
    values[self] = method(self)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 218, in promos
    return MiniSeries.from_data(self._data[LeagueItemData].promos)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 70, in promos
    return MiniSeriesData.from_dto(self._dto["miniSeries"])
KeyError: 'miniSeries'
```
entry.queue - errors
```bash
Traceback (most recent call last):
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 16, in wrapper
    return values[self]
  File "C:\Users\redacted\Anaconda3\lib\weakref.py", line 394, in __getitem__
    return self.data[ref(key)]
KeyError: <weakref at 0x0000022B674E6458; to 'LeagueEntry' at 0x0000022B6750AF98>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\redacted\OneDrive\Bots\BlitzcrankBot\test.py", line 14, in <module>
    print(entry.queue)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 18, in wrapper
    values[self] = method(self)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 202, in queue
    return Queue(self._data[LeagueItemData].queue)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 54, in queue
    return self._dto["queue"]
KeyError: 'queue'
```
entry.region - errors
```bash
Traceback (most recent call last):
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 16, in wrapper
    return values[self]
  File "C:\Users\redacted\Anaconda3\lib\weakref.py", line 394, in __getitem__
    return self.data[ref(key)]
KeyError: <weakref at 0x0000013626CC6458; to 'LeagueEntry' at 0x0000013626CE9FD0>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\redacted\OneDrive\Bots\BlitzcrankBot\test.py", line 15, in <module>
    print(entry.region)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 18, in wrapper
    values[self] = method(self)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 189, in region
    return Region(self._data[LeagueItemData].region)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 46, in region
    return self._dto["region"]
KeyError: 'region'
```
entry.tier - errors
```bash
Traceback (most recent call last):
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 16, in wrapper
    return values[self]
  File "C:\Users\redacted\Anaconda3\lib\weakref.py", line 394, in __getitem__
    return self.data[ref(key)]
KeyError: <weakref at 0x0000024B0FC36458; to 'LeagueEntry' at 0x0000024B0FC60048>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\redacted\OneDrive\Bots\BlitzcrankBot\test.py", line 16, in <module>
    print(entry.tier)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\merakicommons\cache.py", line 18, in wrapper
    values[self] = method(self)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 198, in tier
    return Tier(self._data[LeagueItemData].tier)
  File "C:\Users\redacted\Anaconda3\lib\site-packages\cassiopeia\core\league.py", line 50, in tier
    return self._dto["tier"]
KeyError: 'tier'
```
entry.veteran - works
entry.wins - works

tl;dr

Works | Does not work
-------|----------------
entry.division | entry.fresh_blood
entry.hot_streak | entry.name
entry.league_points | entry.platform
entry.losses | entry.promos
entry.veteran | entry.queue
entry.wins | entry.region
\- | entry.tier

**Comment by jjmaldonis** [comment]
It looks like some of the `League` attributes got mixed into the `LeagueEntry` class. `name` and `queue` shouldn't be there, and the others we should either inherit from the parent `League` class (with two exceptions, see below).

`promos` / `miniSeries` probably isn't always returned from the API, so we should add that in as empty (in the riotapi datasource) if it isn't provided from Riot.

`fresh_blood` not working is another bug -- I think the `_dto["fresh_blood"]` just needs to be changed to `_dto["freshBlood"]` and it will work though.

**Comment by jjmaldonis** [comment]
fixed by ad65b841a061002d46dc5a3d6d9e17aa6e9591ba
